types.go

```diff
@@ -226,6 +230,8 @@ type TableMeta interface {
 	secondary() map[string]anyIndexer      // Secondary indexers (if any)
 	sortableMutex() internal.SortableMutex // The sortable mutex for locking the table for writing
 	anyChanges(txn WriteTxn) (anyChangeIterator, error)
+	proto() any
+	unmarshalYAML(data string) (any, error)
```

Comment:
should we add this here, part of the TableMeta as it's only usage will be part of this `script.go`, or extract to a different package/even the scriptutils.

Reply:
Yeah, I'll need to revisit these weird ones a bit and see how I can mostly hide them. They do need to be in the `genTable` though as that's the place where we know what `Obj` is.

This one was a bit annoying as I need to call `yaml.Unmarshal` with a concrete thing and if it goes via the `any` it'll just end up producing `map[string]any` as a result. Will have to play around with it.

Reply:
Yeah I don't think there's a way around having `unmarshalYAML` here. We need to call `yaml.Unmarshal` on the concrete type as `Unmarshal` reflects on the type of the destination value and if we go via `any` we lose the type information at runtime. What I'll do though is split of the weird internal things into `tableInternal` interface so it's not so visible.